tests/e2e/test_operator.py

```diff
@@ -2262,6 +2263,221 @@ def test_032(self):
+        util.restart_operator()
+        out = kubectl.launch("get pods -l app=clickhouse-operator", ns=settings.operator_namespace).splitlines()[1]
+        operator_pod = re.split(r'[\t\r\n\s]+', out)[0]
+
+    with Then("check for `chi_clickhouse_metric_fetch_errors` string with zero value `1` at the end and delete the chi"):
+        check_metrics_monitoring(operator_namespace, operator_pod, expect_pattern="^chi_clickhouse_metric_fetch_errors{(.*?)} 0$")
+        kubectl.delete_chi(chi)
+
+
+@TestScenario
+@Requirements(RQ_SRS_026_ClickHouseOperator_CHI_ConnectWithHttps("1.0"))
+@Name("test_034. Check CHI HTTPS connection from local client")
+def test_034(self):
+    """Check ClickHouse server can be deployed by ClickHouse Operator with the support for `HTTPS` connection
+    by creating a ClickHouse installation with HTTPS enabled and executing a simple query from a local ClickHouse client 
+    with the`--secure` option when port forwarding is enabled.
+    """
+    self.context.shell = Shell()
+
+    try:
+        with Given("clickhouse-operator pod exists"):
+            kubectl.wait_field("pods", util.operator_label, ".status.containerStatuses[*].ready", "true,true",
+                            ns=settings.operator_namespace)
+            assert kubectl.get_count("pod", ns='--all-namespaces', label=util.operator_label) > 0, error()
+            out = kubectl.launch("get pods -l app=clickhouse-operator", ns=settings.operator_namespace).splitlines()[1]
+
+        with When("create the chi with secure connection"):
+            manifest = "manifests/chi/test-034-https-check-secure.yaml"
+            chi = yaml_manifest.get_chi_name(util.get_full_path(manifest))
+
+            kubectl.create_and_check(
+                manifest=manifest,
+                check={
+                    "apply_templates": {
+                        settings.clickhouse_template,
+                        "manifests/chit/tpl-persistent-volume-100Mi.yaml",
+                    },
+                    "object_counts": {
+                        "statefulset": 1,
+                        "pod": 1,
+                        "service": 2,
+                    },
+                    "do_not_delete": 1,
+                },
+                timeout=600,
+            )
+            kubectl.create_and_check(
+            manifest=manifest,
+            check={
+                "pod_count": 1,
+                "pod_volumes": {
+                    "/var/lib/clickhouse",
+                },
+                "do_not_delete": 1,
+            },
+            timeout=1200,
+            )
+            
+            kubectl.wait_jsonpath("pod", "chi-test-034-https-check-secure-t1-0-0-0", "{.status.containerStatuses[0].ready}", "true",
+                                ns=kubectl.namespace)
+
+        with When("I enable port forwarding"):
+            self.context.shell("kubectl port-forward -n test pod/chi-test-034-https-check-secure-t1-0-0-0 9440:9440 --address 0.0.0.0 &")
```

Comment:
We shall use `f"kubectl port-forward -n {settings.test_namespace} service/clickhouse-test-034-https-check-serure"` to test service with `type: LoadBalancer` instead of test pod/*`

clients outside kubernetes cluster will connect to LoadBalancer service by default

Reply:
Hi, I had two services running:
```
NAME                                        TYPE           CLUSTER-IP     EXTERNAL-IP   PORT(S)                         AGE
chi-test-operator-https-connection-t1-0-0   ClusterIP      None           <none>        8123/TCP,9000/TCP,9009/TCP      40s
clickhouse-test-operator-https-connection   LoadBalancer   10.97.238.72   <pending>     8123:32582/TCP,9000:31472/TCP   58s
```

I tried to forward the host TCP port 9440 to available TCP port of the service :
```
kubectl port-forward -n test service/clickhouse-test-operator-https-connection 9440:9000 --address 0.0.0.0 
Forwarding from 0.0.0.0:9440 -> 9000
Handling connection for 9440
E0728 18:41:02.281488  879940 portforward.go:385] error copying from local connection to remote stream: read tcp4 127.0.0.1:9440->127.0.0.1:60974: read: connection reset by peer
Handling connection for 9440

```
But the Clickhouse-server connection failed:
```
clickhouse-client --secure --host=127.0.0.1 --port 9440 
ClickHouse client version 22.2.2.1.
Connecting to 127.0.0.1:9440 as user default.
Code: 210. DB::NetException: SSL Exception: error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER (127.0.0.1:9440). (NETWORK_ERROR)

```

Reply:
ok. in this case we need `serviceTemplate` with 9440 for `type: LoadBalancer`

Reply:
Ok, I will try creating it and check the connectivity. It will help to learn something new.

Reply:
Hi, I deployed `serviceTemplate` with TCP 9440, and port-forwad to the service :
```
kubectl port-forward -n test service/clickhouse-test-operator-https-connection 9440:9440 --address 0.0.0.0 
Forwarding from 0.0.0.0:9440 -> 9440
```
Then connect is successful : 
```
clickhouse-client --secure --host=127.0.0.1 --port 9440
ClickHouse client version 22.2.2.1.
Connecting to 127.0.0.1:9440 as user default.
Connected to ClickHouse server version 22.3.10 revision 54455.

ClickHouse client version is older than ClickHouse server. It may lack support for new features.

Warnings:
 * Effective user of the process (clickhouse) does not match the owner of the data (root).

chi-test-operator-https-connection-t1-0-0-0.chi-test-operator-https-connection-t1-0-0.test.svc.cluster.local :) 
```
I have updated the PR. Thanks :)